centreon/src/Core/Security/Token/Infrastructure/Repository/DbReadTokenRepository.php

```diff
@@ -164,6 +164,35 @@ public function existsByNameAndUserId(string $tokenName, int $userId): bool
         return (bool) $statement->fetchColumn();
     }
 
+    /**
+     * @param string $token
+     *
+     * @return bool
+     */
+    public function isTokenTypeManual(string $token): bool
+    {
+        try {
+            $statement = $this->db->prepare(
+            $this->translateDbName(
+                'SELECT token_type FROM `:db`.security_authentication_tokens WHERE token = :token'
+            )
+            );
+
+            $statement->bindValue(':token', $token, \PDO::PARAM_STR);
+            $statement->execute();
+            $result = $statement->fetch(\PDO::FETCH_ASSOC);
+
+            return $result && $result['token_type'] === self::TYPE_MANUAL;
```

Comment:
you could also add a where clause on your token_type, so you just have to SELECT 1  et return ! empty($result);

if result is empty you will return false (your token is not found as manual), if result is not empty you will return true (your token is found as manual)

```suggestion
            $this->translateDbName(
                'SELECT 1 FROM `:db`.security_authentication_tokens WHERE token = :token AND token_type = :tokenType'
            )
            );

            $statement->bindValue(':token', $token, \PDO::PARAM_STR);
            $statement->bindValue(':tokenType',  self::TYPE_MANUAL, \PDO::PARAM_STR);
            $statement->execute();
            $result = $statement->fetch(\PDO::FETCH_ASSOC);

            return ! empty($result);
```